pkg/virtctl/credentials/add-key/add-key_test.go

```diff
@@ -0,0 +1,562 @@
+			Spec: vmi.Spec,
+		}
+
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      secretName,
+				Namespace: metav1.NamespaceDefault,
+				OwnerReferences: []metav1.OwnerReference{{
+					APIVersion: v1.VirtualMachineGroupVersionKind.GroupVersion().String(),
+					Kind:       v1.VirtualMachineGroupVersionKind.Kind,
+					Name:       vm.Name,
+					UID:        vm.UID,
+					Controller: pointer.Bool(true),
+				}},
+			},
+		}
+
+		_, err := kubeClient.CoreV1().Secrets(metav1.NamespaceDefault).Create(context.Background(), secret, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		ctrl := gomock.NewController(GinkgoT())
+		kubecli.GetKubevirtClientFromClientConfig = kubecli.GetMockKubevirtClientFromClientConfig
+		kubecli.MockKubevirtClientInstance = kubecli.NewMockKubevirtClient(ctrl)
+
+		vmiInterface := kubecli.NewMockVirtualMachineInstanceInterface(ctrl)
+		vmInterface := kubecli.NewMockVirtualMachineInterface(ctrl)
+
+		kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachineInstance(metav1.NamespaceDefault).Return(vmiInterface).AnyTimes()
+		kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(metav1.NamespaceDefault).Return(vmInterface).AnyTimes()
+		kubecli.MockKubevirtClientInstance.EXPECT().CoreV1().Return(kubeClient.CoreV1()).AnyTimes()
+
+		vmiInterface.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(func(_ any, name string, _ any) (*v1.VirtualMachineInstance, error) {
+			if name == vmName && vmi != nil {
+				return vmi, nil
+			}
+			return nil, errors.NewNotFound(schema.GroupResource{
+				Group:    core.GroupName,
+				Resource: "VirtualMachineInstance",
+			}, name)
+		}).AnyTimes()
+
+		vmInterface.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(func(_ any, name string, _ any) (*v1.VirtualMachine, error) {
+			if name == vmName && vm != nil {
+				return vm, nil
+			}
+			return nil, errors.NewNotFound(schema.GroupResource{
+				Group:    core.GroupName,
+				Resource: "VirtualMachine",
+			}, name)
+		}).AnyTimes()
+
+		vmInterface.EXPECT().Patch(gomock.Any(), gomock.Any(), types.JSONPatchType, gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(func(_ any, name string, _ any, patchData []byte, _ any, _ ...any) (*v1.VirtualMachine, error) {
+			if name != vmName || vm == nil {
+				return nil, errors.NewNotFound(schema.GroupResource{
+					Group:    core.GroupName,
+					Resource: "VirtualMachine",
+				}, name)
+			}
+
+			patch, err := jsonpatch.DecodePatch(patchData)
+			if err != nil {
+				return nil, err
+			}
+
+			vmJson, err := json.Marshal(vm)
+			if err != nil {
+				return nil, err
+			}
+
+			modifiedVmJson, err := patch.Apply(vmJson)
+			if err != nil {
+				return nil, err
+			}
+
+			err = json.Unmarshal(modifiedVmJson, vm)
+			if err != nil {
+				return nil, err
+			}
+
+			return vm, nil
+		}).AnyTimes()
+
+	})
+
+	It("should fail if no key is specified", func() {
```

Comment:
This and the next 3 tests could be reduced to a Table with an array of commands and error message

Reply:
True, but I don't think it would be simpler, or less code.